text/0096-contract-metadata.md

```diff
@@ -0,0 +1,213 @@
+# Casper On-Chain Contract Metadata Standard
+
+## Summary
+
+[summary]: #summary
+
+This standard proposes a unified way for Casper smart contracts to self-describe with on-chain metadata, improving contract discoverability and trust. It defines optional metadata fields – `name`, `description`, `icon_uri`, and `project_uri` – which a contract can expose via dedicated entry points. These fields allow a contract to broadcast its human-readable name and purpose, along with links to an icon image and project website.
+
+All metadata is **immutable** (set once at deployment and never altered), ensuring consistency and integrity.
+
+By adopting this standard, Casper developers enable wallets, network explorers, and indexers to fetch and display contract information directly from the blockchain, much like how token standards (e.g. CEP-18 for fungible tokens or CEP-95 for NFTs) provide on-chain names and symbols, underscoring the community’s move toward rich on-chain identification.
+
+**Metadata Fields**:
+
+- `name`: A human-readable name for the contract (e.g. "Casper DEX").
+- `description`: A brief description of the contract’s purpose or functionality.
+- `icon_uri`: A URI (URL) pointing to an icon image for the contract (e.g. a logo).
+- `project_uri`: A URI pointing to the contract’s project website or documentation.
```

Comment:
From the point of view of smart contract developer, I'd like to add.
```suggestion
- `project_uri`: A URI pointing to the contract’s project website or documentation.
- `toolchain`: A detailed info about how contract was build. Like: `rustc 1.81.0 (eeb90cda1 2024-09-04), odra 1.5.1`.
- `event_standard`: `ces 0.1` or `native`
- `git_hash`: Exact hash of the repository, the version of the contract is coming from.
```

Reply:
I think toolchain should be placeholder for identification of how contracts are build. If you develop in `go`, then put some compiler info.

Reply:
I'd have all fields mandatory and have a value that says no events are used line `none`.